### ISSUE TITLE
react-redux libdef broken for mapState factory signature

### ISSUE THREAD

**Issue by B-Reif** [flowtype/flow-typed#2109]
From the react-redux docs:

> Note: in advanced scenarios where you need more control over the rendering performance, mapStateToProps() can also return a function. In this case, that function will be used as mapStateToProps() for a particular component instance. This allows you to do per-instance memoization.

I recently updated my flow-typed installations and now it's breaking on every instance of this.

```js
const makeMapStateToProps = (): * => {
	const selector = makeSelector();
	const mapStateToProps = (state, ownProps) => {
		const { foo } = selector(state, ownProps.id);
		return { foo };
	};
	return mapStateToProps; // Error! Cannot return `mapStateToProps` because function is not an object.
};

export default connect(makeMapStateToProps)(MyComponent);
```

I use this signature extensively to prevent needless re-renders, and now flow is complaining about them everywhere.

**Comment by lukeapage** [comment]
Duplicate of #1924 and fixed by #2105

**Comment by AndrewSouthpaw** [comment]
Closing because it's a dupe.
